File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;
        HashSet<Integer> hset = new HashSet<>();
        for(int num : nums){
            hset.add(num);
        }
        int maxlen = 0;
        for(int num : hset){
            if(!hset.contains(num-1)){
                int curr = num;
                int len = 0;
                while(hset.contains(curr)){
                    curr++;
                    len++;
                }
                maxlen = Math.max(maxlen, len);
            }
        }
        
        return maxlen;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        set<int> s(nums.begin(), nums.end());
        
        int maxlen = 1;
        for(int x : s){
            if(s.find(x-1) == s.end()){
                int len = 0;
                int curr = x;
                while(s.find(curr) != s.end()){
                    len++;
                    curr++;
                }
                maxlen = max(maxlen, len);
            }
        }
        
        return maxlen ;
    }
};